src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/VariantRecalibrator.java

```diff
@@ -0,0 +1,852 @@
+
+    @Override
+    public void onTraversalStart() {
+        dataManager = new VariantDataManager( new ArrayList<>(USE_ANNOTATIONS), VRAC );
+
+        if (RSCRIPT_FILE != null && !RScriptExecutor.RSCRIPT_EXISTS)
+            Utils.warnUser(logger, String.format(
+                    "Rscript not found in environment path. %s will be generated but PDF plots will not.",
+                    RSCRIPT_FILE));
+
+        if( IGNORE_INPUT_FILTERS != null ) {
+            ignoreInputFilterSet.addAll( IGNORE_INPUT_FILTERS );
+        }
+
+        try {
+            tranchesStream = new PrintStream(TRANCHES_FILE);
+        } catch (FileNotFoundException e) {
+            throw new UserException.CouldNotCreateOutputFile(TRANCHES_FILE, e);
+        }
+
+        for( FeatureInput<VariantContext> variantFeature : resource ) {
+            dataManager.addTrainingSet( new TrainingSet( variantFeature ) );
+        }
+
+        if( !dataManager.checkHasTrainingSet() ) {
+            throw new UserException.CommandLineException(
+                    "No training set found! Please provide sets of known polymorphic loci marked with the training=true ROD binding tag. For example, -resource:hapmap,VCF,known=false,training=true,truth=true,prior=12.0 hapmapFile.vcf" );
+        }
+        if( !dataManager.checkHasTruthSet() ) {
+            throw new UserException.CommandLineException(
+                    "No truth set found! Please provide sets of known polymorphic loci marked with the truth=true ROD binding tag. For example, -resource:hapmap,VCF,known=false,training=true,truth=true,prior=12.0 hapmapFile.vcf" );
+        }
+
+        //TODO: this should be refactored/consolidated as part of
+        // https://github.com/broadinstitute/gatk/issues/121 and
+        // https://github.com/broadinstitute/gatk/issues/1116
+        // Initialize VCF header lines
+        final Set<VCFHeaderLine> hInfo = new HashSet<>();
+        hInfo.add(new VCFHeaderLine("source", this.getClass().getSimpleName()));
+        VariantRecalibrationUtils.addVQSRStandardHeaderLines(hInfo);
+        Set<VCFHeaderLine> actualLines = null;
+        SAMSequenceDictionary sequenceDictionary = null;
+        if (hasReference()) {
+            File refFile = referenceArguments.getReferenceFile();
+            sequenceDictionary= this.getReferenceDictionary();
+            actualLines = VcfUtils.updateHeaderContigLines(hInfo, refFile, sequenceDictionary, true);
+        }
+        else {
+            sequenceDictionary = getHeaderForVariants().getSequenceDictionary();
+            if (null != sequenceDictionary) {
+                actualLines = VcfUtils.updateHeaderContigLines(hInfo, null, sequenceDictionary, true);
+            }
+            else {
+                actualLines = hInfo;
+            }
+        }
+        //TODO: integrate lenient argument
+        recalWriter = GATKVariantContextUtils.createVCFWriter(new File(output), getBestAvailableSequenceDictionary(), true);
```

Comment:
Will the sequence dictionary selected above match the one returned here by `getBestAvailableSequenceDictionary()`?